### PR TITLE
Migrate to Winterfell release

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/hash/rescue/rpo/mod.rs [hunk 1]
```diff
@@ -164,6 +165,12 @@ impl Hasher for Rpo256 {
         RpoDigest::new(state[DIGEST_RANGE].try_into().unwrap())
     }
 
+    fn merge_many(values: &[Self::Digest]) -> Self::Digest {
+        let elements: Vec<Felt> =
+            Self::Digest::digests_as_elements(values.iter()).copied().collect();
+        Self::hash_elements(&elements)
```
Comment (bobbinth) [inline src/hash/rescue/rpo/mod.rs:171]: Can we avoid allocating vectors here? Maybe we could do something similar to what is done in winterfell? For example, add a new method to `Digest` which would return a slice that can be passed in directly to `Self::hash_elements().`

File: src/hash/blake/mod.rs [hunk 2]
```diff
@@ -114,6 +114,11 @@ impl Hasher for Blake3_256 {
         Self::hash(prepare_merge(values))
     }
 
+    fn merge_many(values: &[Self::Digest]) -> Self::Digest {
+        let bytes: Vec<u8> = values.iter().flat_map(|v| v.as_bytes()).collect();
+        Blake3Digest(blake3::hash(&bytes).into())
```
Comment (bobbinth) [inline src/hash/blake/mod.rs:119]: Similar comment here: would be great to do this without allocating new vectors.

File: src/hash/rescue/rpo/mod.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply src/hash/rescue/rpo/mod.rs]: Makes sense. I have added the same implementations that we have in Winterfell here.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**